packages/browser-wallet/src/popup/pages/Recovery/RecoveryFinish.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useContext } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { useAtomValue, useAtom } from 'jotai';
+import { useTranslation } from 'react-i18next';
+import { credentialsAtom, selectedAccountAtom } from '@popup/store/account';
+import { identitiesAtom } from '@popup/store/identity';
+import Button from '@popup/shared/Button';
+import { absoluteRoutes } from '@popup/constants/routes';
+import { noOp, displayAsCcd } from 'wallet-common-helpers';
+import { displaySplitAddress } from '@popup/shared/utils/account-helpers';
+import { BackgroundResponseStatus, RecoveryBackgroundResponse } from '@shared/utils/types';
+import { fullscreenPromptContext } from '@popup/page-layouts/FullscreenPromptLayout';
+import PageHeader from '@popup/shared/PageHeader';
+
+interface Location {
+    state: {
+        payload: RecoveryBackgroundResponse;
+    };
+}
+
+export default function DisplayRecoveryResult() {
+    const { state } = useLocation() as Location;
+    const { payload } = state;
+    const { setReturnLocation, withClose } = useContext(fullscreenPromptContext);
+    const { t } = useTranslation('recovery');
+    const navigate = useNavigate();
+    const identities = useAtomValue(identitiesAtom);
+    const credentials = useAtomValue(credentialsAtom);
```

Comment:
You can't just show everything in the wallet as being what has been recovered. That only works for an empty wallet.

Reply:
I changed it so the recovery sends which identities/accounts have been recovered to this page, and it only displays what was recovered ( + identities that has accounts added)